packages/headless/src/api/search/search-api-error-response.ts

```diff
@@ -9,3 +9,11 @@ export interface SearchAPIErrorWithStatusCode {
 export interface SearchAPIErrorWithExceptionInBody {
   exception: QueryException;
 }
+
+export function buildDisconnectedError(): SearchAPIErrorWithStatusCode {
+  return {
+    statusCode: 0,
+    type: 'Disconnected',
+    message: 'Could not connect',
+  };
+}
```

Comment:
I'm not sure where this function should belong, if anywhere. In fact, I wonder if it's even appropriate to resolve disconnection errors as fake HTTP errors when no HTTP response was ever received.

On the other hand, the status code 0 seems to exist for these kinds of purposes already.